### PR TITLE
renovate: Schedule a monthly PR for Go dependencies

### PR DESCRIPTION
**Pull request by tsenart**
To be reviewed by @sourcegraph/core-services.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -2,5 +2,14 @@
...
   "semanticCommits": false,
-  "labels": ["bot"]
+  "labels": ["bot"],
+  "packageRules": [
+    {
+      "languages": ["golang"],
+      "packagePatterns": [".*"],
+      "groupName": "godeps",
+      "schedule": ["on the 22nd of each month"],
+      "reviewers": ["team:core-services"]
```
Comment (felixfbecker) [inline renovate.json]: Can you also set `assignees`? It is currently set to Stephen:

https://github.com/sourcegraph/renovate-config/blob/29514e06910cbfd3597169932194e05b773d6bba/renovate.json#L19

File: renovate.json [hunk 2]
```diff
@@ -2,5 +2,14 @@
...
   "extends": ["github>sourcegraph/renovate-config"],
   "semanticCommits": false,
-  "labels": ["bot"]
+  "labels": ["bot"],
+  "packageRules": [
+    {
+      "languages": ["golang"],
+      "packagePatterns": [".*"],
+      "groupName": "godeps",
+      "schedule": ["on the 22nd of each month"],
```
Comment (felixfbecker) [inline renovate.json]: Our general schedule is the first week of each month. Is there a reason to deviate from that? One consideration is that Renovate will limit the amount of PRs open, so if you only have a timeframe of one day, you are likely to not get all updates (until the next month).

File: renovate.json [hunk 3]
```diff
@@ -2,5 +2,14 @@
   "$schema": "http://json.schemastore.org/renovate",
   "extends": ["github>sourcegraph/renovate-config"],
   "semanticCommits": false,
-  "labels": ["bot"]
+  "labels": ["bot"],
+  "packageRules": [
+    {
+      "languages": ["golang"],
+      "packagePatterns": [".*"],
+      "groupName": "godeps",
```
Comment (felixfbecker) [inline renovate.json]: Consider that this means if you don't want to udpate one of the dependencies there is no way to do that without manually changing the PR

**Comment by tsenart** [comment]
> I think this would be best to go into our shared config, so it also applies e.g. to the update PRs in https://github.com/sourcegraph/infrastructure

Oh, I see! I didn't know about that. I'll open this PR there and address your feedback in it. Thanks!

File: renovate.json (hunk 3 above)
Comment (tsenart) [reply renovate.json]: That's what we want. A batch workflow, once a month, one PR.

**Pull request closed without merging**